lib/devise/models/confirmable.rb

```diff
@@ -169,6 +169,7 @@ def skip_reconfirmation!
         # in models to map to a nice sign up e-mail.
         def send_on_create_confirmation_instructions
           send_confirmation_instructions
+          skip_reconfirmation!
```

Comment:
ping @lucasmazza 
In short there was breaking change released in 4.0.1: when updating email from `nil` it is saved to `unconfirmed email`, before this PR it was saved to `email`. Should this be fixed?